File: pureabs.java
import java.util.Scanner;

interface Area {
    void displayArea();
}

class Square implements Area {
    private float side;

    Square(float side) {
        this.side = side;
    }

    @Override
    public void displayArea() {
        System.out.println("Area of square: " + (side * side));
    }
}

class Rectangle implements Area {
    private float length;
    private float breadth;

    Rectangle(float length, float breadth) {
        this.length = length;
        this.breadth = breadth;
    }

    @Override
    public void displayArea() {
        System.out.println("Area of rectangle: " + (length * breadth));
    }
}

class Circle implements Area {
    private float radius;

    Circle(float radius) {
        this.radius = radius;
    }

    @Override
    public void displayArea() {
        System.out.println("Area of circle: " + (Math.PI * radius * radius));
    }
}

public class ParAbsInterface {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        float side, length, breadth, radius;

        System.out.print("Enter the side of a square: ");
        side = sc.nextFloat();
        Square square = new Square(side);
        System.out.print("Area of square: ");
        square.displayArea();

        System.out.print("Enter the length and breadth of a rectangle: ");
        length = sc.nextFloat();
        breadth = sc.nextFloat();
        Rectangle rectangle = new Rectangle(length, breadth);
        System.out.print("Area of rectangle: ");
        rectangle.displayArea();

        System.out.print("Enter the radius of a circle: ");
        radius = sc.nextFloat();
        Circle circle = new Circle(radius);
        System.out.print("Area of circle: ");
        circle.displayArea();

        sc.close();
    }
}

File: pureabs.cpp
#include <iostream>

// Abstract base class Shape
class Shape {
public:
    // Pure virtual function for calculating area
    virtual void area() const = 0;
};

// Concrete class Circle derived from Shape
class Circle : public Shape {
private:
    double radius;
public:
    Circle(double r) : radius(r) {}

    // Implementation of the area() function for Circle
    void area() const override {
        std::cout << "Area of circle: " << 3.14159 * radius * radius << std::endl;
    }
};

// Concrete class Rectangle derived from Shape
class Rectangle : public Shape {
private:
    double length;
    double breadth;
public:
    Rectangle(double l, double b) : length(l), breadth(b) {}

    // Implementation of the area() function for Rectangle
    void area() const override {
        std::cout << "Area of rectangle: " << length * breadth << std::endl;
    }
};

int main() {
    double radius, length, breadth;

    std::cout << "Enter the radius of a circle: ";
    std::cin >> radius;
    Circle circle(radius);

    std::cout << "Enter the length and breadth of a rectangle: ";
    std::cin >> length >> breadth;
    Rectangle rectangle(length, breadth);

    // Using the pure abstract class to call area function
    const Shape* shapes[2] = {&circle, &rectangle};

    // Calling the area function for each shape
    for (const Shape* shape : shapes) {
        shape->area();
    }

    return 0;
}
